### PR TITLE
Revamp command bar usages

### PR DESCRIPTION
**Pull request by aekt**
ref DEV-1753

### CODE REVIEW

**Comment by aekt** [comment]
Seems most usages of `<CommandBarContainer>` are together with `<FormContainer>`, will apply this bottom bar container in DEV-1754

**Comment by louischan-oursky** [comment]
We usually do not break down PR like this small. Expect this PR to have actual usage of this component. Also, please provide screenshots to show the changes. (something that you can only do if this PR does include actual usage of this component)

**Comment by tung2744** [comment]
I suggest you can put the component in `StoryBookScreen` so that we can test it.

**Comment by louischan-oursky** [comment]
TIL There is a story book screen.

**Comment by aekt** [comment]
@tung2744 ready for review now :adore2x:

One of the screen ("Authentication" > "Anonymous Users") I'll revisit in DEV-1790, I now place the "Save" button at the bottom first :adore2x:

image

File: portal/src/BottomActionContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+export interface BottomActionContainerProps {
+  position: "sticky" | "end";
+  className?: string;
+  isLoading?: boolean;
+  messageBar?: React.ReactNode;
+  hideActionButtons?: boolean;
+  bottomContent?: React.ReactNode;
+  renderHeaderContent?: (
+    defaultHeaderContent: React.ReactNode
+  ) => React.ReactNode;
```
Comment (louischan-oursky) [inline portal/src/BottomActionContainer.tsx:22]: I prefer this, looks more composable.

```tsx
interface BottomActionContainerHeaderComponentProps {
  children?: React.ReactNode;
}

interface BottomActionContainerProps {
  HeaderComponent?: React.Component<BottomActionContainerHeaderComponentProps>;
}

function BottomActionContainer(props) {
  const { HeaderComponent } = props;
  let header = defaultHeaderContent;
  if (HeaderComponent != null) {
    header = <HeaderComponent>{header}</HeaderComponent>
  }
}

```

File: portal/src/BottomActionContainer.module.css [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+.header {
+}
+
+.contentExpand {
+  @apply flex-1;
+}
+
+.footer {
+}
```
Comment (louischan-oursky) [inline portal/src/BottomActionContainer.module.css:15]: The <div class="header"> and <div class="footer"> seems redundant. Or you document a reason to have them?

**Review by louischan-oursky — CHANGES_REQUESTED** [review]

**Comment by tung2744** [comment]
Not sure if we should create a feature branch for the portal updates, so we do not deploy these changes before it is complated?

File: portal/src/BottomActionContainer.module.css (hunk 2 above)
Comment (aekt) [reply portal/src/BottomActionContainer.module.css]: Right, just removed

File: portal/src/BottomActionContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  className?: string;
+  isLoading?: boolean;
+  messageBar?: React.ReactNode;
+  hideActionButtons?: boolean;
+  bottomContent?: React.ReactNode;
+  HeaderComponent?: React.VFC<BottomActionContainerHeaderComponentProps>;
+  children?: React.ReactNode;
+}
+
+const BottomActionContainer: React.VFC<BottomActionContainerProps> =
```
Comment (louischan-oursky) [inline portal/src/BottomActionContainer.tsx:27]: Rename this to `DefaultLayout`

**Review by louischan-oursky — CHANGES_REQUESTED** [review]

**Comment by louischan-oursky** [comment]
There are still some type errors.

File: portal/src/graphql/portal/Web3ConfigurationScreen.tsx [hunk 4]
```diff
@@ -494,6 +494,7 @@ const Web3ConfigurationContent: React.VFC<Web3ConfigurationContentProps> =
           <ScreenDescription className={styles.widget}>
             <FormattedMessage id="Web3ConfigurationScreen.description" />
           </ScreenDescription>
+          <Widget className={styles.widget} />
```
Comment (louischan-oursky) [inline portal/src/graphql/portal/Web3ConfigurationScreen.tsx:497]: Intentional?

File: portal/src/FormContainer.tsx [hunk 5]
```diff
@@ -117,18 +76,46 @@ const FormContainer_: React.VFC<FormContainerProps> = function FormContainer_(
...
-      <CommandBarContainer
-        renderHeaderContent={renderHeaderContent}
-        hideCommandBar={hideCommandBar}
-        isLoading={isUpdating}
-        primaryItems={items}
-        secondaryItems={farItems}
+      <DefaultLayout
+        position={stickyFooterComponent ? "sticky" : "end"}
+        HeaderComponent={HeaderComponent}
+        isLoading={false}
```
Comment (louischan-oursky) [inline portal/src/FormContainer.tsx:82]: The only use of DefaultLayout is here, and EditPictureScreen. Does EditPictureScreen need to use `isLoading`? If not, then we should remove isLoading from DefaultLayout.

**Review by louischan-oursky — CHANGES_REQUESTED** [review]

File: portal/src/graphql/portal/Web3ConfigurationScreen.tsx (hunk 4 above)
Comment (aekt) [reply portal/src/graphql/portal/Web3ConfigurationScreen.tsx:497]: Not intentional, thanks for catching

**Comment by louischan-oursky** [comment]
eslint failed.

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**